### PR TITLE
Enable CodeQL code scanning as a GitHub Action

### PR DESCRIPTION
**Pull request by betatim**
This GitHub Action uses CodeQL to scan our code for security problems.

I'd be interested to see how well it works (or not). Can we enable it for a while to see if we like it or find it annoying/noisy?

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+name: "CodeQL"
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [master]
+  schedule:
+    - cron: '29 13 * * 5'
```
Comment (consideRatio) [inline .github/workflows/codeql-analysis.yml:15]:
```suggestion
    # At 13:29 on Friday - https://crontab.guru/
    - cron: '29 13 * * 5'
```

**Comment by consideRatio** [comment]
LGTM, will merge following CI tests completed. I added a comment to help quickly understand the cron specification.

**Comment by consideRatio** [comment]
@betatim :'( it seems like we have an unrelated CI failure, where a pip install -r /tmp/requirements.txt fail, because PyYaml dependency yarl fails to install, because gcc fails to build something...

Unrelated I assume right? I dare not merge myself, but I +1 to merge if you also think it is unrelated and safe to merge.

**Comment by betatim** [comment]
I'll investigate the build failure. Unrelated to the PR but if we can't build the hub image then that is a problem that needs fixing :D

**Pull request merged**